pkg/cli/cli.go

```diff
@@ -176,8 +178,19 @@ func NewOcCommand(o kubecmd.KubectlOptions) *cobra.Command {
 		Long:  cliLong,
 		Run:   runHelp,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			rest.SetDefaultWarningHandler(warningHandler)
-
+			switch cmd.Name() {
+			case "status":
+				rest.SetDefaultWarningHandler(apps.NewIgnoreDeploymentConfigWarningHandler(warningHandler))
+			case "get":
+				if slices.ContainsFunc(args, func(s string) bool { return slices.Contains(strings.Split(s, ","), "all") }) {
+					rest.SetDefaultWarningHandler(apps.NewIgnoreDeploymentConfigWarningHandler(warningHandler))
```

Comment:
I'm not sure this will work because as we concluded that builder does not take custom warning handler into account https://github.com/kubernetes/kubectl/issues/1586#issuecomment-2102450494 and `get` uses builder or our conclusion is wrong?